src/clj_http/conn_mgr.clj

```diff
@@ -6,53 +6,32 @@
            (org.apache.http.conn.socket PlainConnectionSocketFactory)
-           (org.apache.http.impl.conn BasicClientConnectionManager
-                                      PoolingClientConnectionManager
-                                      SchemeRegistryFactory
-                                      SingleClientConnManager)
+           (org.apache.http.impl.conn PoolingClientConnectionManager)
            (org.apache.http.impl.conn BasicHttpClientConnectionManager
-                                      PoolingHttpClientConnectionManager)))
-
-(def ^SSLSocketFactory insecure-socket-factory
-  (SSLSocketFactory. (reify TrustStrategy
-                       (isTrusted [_ _ _] true))
-                     (reify X509HostnameVerifier
-                       (^void verify [this ^String host ^SSLSocket sock]
-                        ;; for some strange reason, only TLSv1 really
-                        ;; works here, if you know why, tell me.
-                        (.setEnabledProtocols
-                         sock (into-array String ["TLSv1"]))
-                        (.setWantClientAuth sock false)
-                        (let [session (.getSession sock)]
-                          (when-not session
-                            (.startHandshake sock))
-                          (aget (.getPeerCertificates session) 0)
-                          ;; normally you'd want to verify the cert
-                          ;; here, but since this is an insecure
-                          ;; socketfactory, we don't
-                          nil))
-                       (^void verify [_ ^String _ ^X509Certificate _]
-                        nil)
-                       (^void verify [_ ^String _ ^"[Ljava.lang.String;" _
-                                      ^"[Ljava.lang.String;" _]
-                        nil)
-                       (^boolean verify [_ ^String _ ^SSLSession _]
-                        true))))
-
-(def ^SSLSocketFactory secure-ssl-socket-factory
-  (doto (SSLSocketFactory/getSocketFactory)
-    (.setHostnameVerifier SSLSocketFactory/STRICT_HOSTNAME_VERIFIER)))
+                                      PoolingHttpClientConnectionManager)
+           (org.apache.http.conn.ssl SSLConnectionSocketFactory
+                                     DefaultHostnameVerifier
+                                     NoopHostnameVerifier
+                                     TrustStrategy)
+           (org.apache.http.ssl SSLContexts)
+           (org.apache.http.pool ConnPoolControl)))
+
+(def ^SSLConnectionSocketFactory insecure-socket-factory
+  (SSLConnectionSocketFactory.
+   (-> (SSLContexts/custom)
+       (.loadTrustMaterial nil (reify TrustStrategy
+                                 (isTrusted [_ _ _] true)))
+       (.build))
+   NoopHostnameVerifier/INSTANCE))
+
+(def ^SSLConnectionSocketFactory secure-ssl-socket-factory
+  (SSLConnectionSocketFactory/getSocketFactory))
```

Comment:
I think this misses the strict hostname verifier I added:

```clojure
(def ^SSLSocketFactory secure-ssl-socket-factory
  (doto (SSLSocketFactory/getSocketFactory)
    (.setHostnameVerifier SSLSocketFactory/STRICT_HOSTNAME_VERIFIER)))
```

(The equivalent for the SSLConnectionSocketFactory version at least)